telemetry/client.go

```diff
@@ -0,0 +1,51 @@
+package telemetry
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/transport"
+)
+
+type Client struct {
+	serverURL  string
+	httpClient *http.Client
+	logger     *zap.Logger
+	keyUID     string
+}
+
+func NewClient(logger *zap.Logger, serverURL string, keyUID string) *Client {
+	return &Client{
+		serverURL:  serverURL,
+		httpClient: &http.Client{Timeout: time.Duration(1) * time.Second},
+		logger:     logger,
+		keyUID:     keyUID,
+	}
+}
+
+func (c *Client) PushReceivedMessages(messages map[transport.Filter][]*types.Message) {
+	c.logger.Debug("Pushing received messages to telemetry server")
+	url := fmt.Sprintf("%s/received-messages", c.serverURL)
+	for filter, messages := range messages {
+		for _, message := range messages {
+			postBody, _ := json.Marshal(map[string]interface{}{
+				"chatId":         filter.ChatID,
+				"messageHash":    string(message.Hash),
```

Comment:
I'd check that `string(message.Hash)` gives us the `0x...` representation, as that's the one we probably want as "canonical"

Reply:
We probably want to have both, you can use `types.EncodeToHex(message.Hash)` to have the hex encoded representation.
Pushing after handle message is useful as well, thought we might drop them in between :) but that's unlikely, in general for the first pass I would push it after, so we can log the application message ID as well

Reply:
Ok cool, this is change has been made, we now log the message id (kept the hash) and also the node name in case of mobile/desktop running for the same account